### PR TITLE
Refactor ModulesDataView

### PR DESCRIPTION
**Pull request by dimitry-**
1. Remove dependency on Process
2. Remove unused code
3. Rename m_FoundPdb to loadable_ and remove unused IsDll method from
   Module

Test: build, run orbit load modules in Modules view.

### CODE REVIEW

File: OrbitGl/ModulesDataView.cpp [hunk 1]
```diff
@@ -105,31 +104,26 @@ void ModulesDataView::DoSort() {
...
     } else {
       enable_verify = true;
     }
   } else {
-    for (int index : a_SelectedIndices) {
+    for (int index : selected_indices) {
       std::shared_ptr<Module> module = GetModule(index);
-      if (!module->GetLoaded()) {
-        if (module->m_FoundPdb) {
+      if (!module->IsLoaded()) {
```
Comment (dpallotti) [inline OrbitGl/ModulesDataView.cpp:125]: `if (module->IsLoadable() && !module->IsLoaded())`?

File: OrbitGl/ModulesDataView.cpp [hunk 2]
```diff
@@ -105,31 +104,26 @@ void ModulesDataView::DoSort() {
...
   bool enable_load = false;
-  bool enable_dll = false;
   bool enable_verify = false;
-  if (a_SelectedIndices.size() == 1) {
-    std::shared_ptr<Module> module = GetModule(a_SelectedIndices[0]);
-    if (!module->GetLoaded()) {
-      if (module->m_FoundPdb) {
+  if (selected_indices.size() == 1) {
+    std::shared_ptr<Module> module = GetModule(selected_indices[0]);
+    if (!module->IsLoaded()) {
```
Comment (dpallotti) [inline OrbitGl/ModulesDataView.cpp:115]: The if-else here is a bit confusing. What about
```
if (module->IsLoadable() && !module->IsLoaded()) {
  enable_load = true;
}
if (module->IsLoaded()) {
  enable_verify = true;
}
```
?

File: OrbitGl/ModulesDataView.h [hunk 3]
```diff
@@ -16,26 +16,33 @@ class ModulesDataView : public DataView {
...
   void OnTimer() override;
   bool WantsDisplayColor() override { return true; }
-  bool GetDisplayColor(int /*a_Row*/, int /*a_Column*/, unsigned char& /*r*/,
-                       unsigned char& /*g*/, unsigned char& /*b*/) override;
+  bool GetDisplayColor(int row, int column, unsigned char& red,
+                       unsigned char& green, unsigned char& blue) override;
   std::string GetLabel() override { return "Modules"; }
 
-  void SetProcess(const std::shared_ptr<Process>& a_Process);
+  void SetData(uint32_t process_id,
```
Comment (dpallotti) [inline OrbitGl/ModulesDataView.h:30]: Maybe `SetModules`?

File: OrbitGl/ModulesDataView.cpp (hunk 1 above)
Comment (dimitry-) [reply OrbitGl/ModulesDataView.cpp]: done

File: OrbitGl/ModulesDataView.cpp (hunk 2 above)
Comment (dimitry-) [reply OrbitGl/ModulesDataView.cpp]: done - also removed top-level if - there is no need to check size of selected indices.

File: OrbitGl/ModulesDataView.h (hunk 3 above)
Comment (dimitry-) [reply OrbitGl/ModulesDataView.h]: done.

**Pull request merged**